daemon/src/collab_settlement/taker.rs

```diff
@@ -0,0 +1,106 @@
+use crate::collab_settlement::protocol::*;
+use crate::command;
+use anyhow::Context;
+use anyhow::Result;
+use async_trait::async_trait;
+use model::libp2p::PeerId;
+use model::OrderId;
+use model::Price;
+use tokio_tasks::Tasks;
+use xtra::Address;
+use xtra_libp2p::Endpoint;
+use xtra_productivity::xtra_productivity;
+
+pub struct Actor {
+    endpoint: Address<Endpoint>,
+    tasks: Tasks,
+    executor: command::Executor,
+    n_payouts: usize,
+}
+
+impl Actor {
+    pub fn new(endpoint: Address<Endpoint>, executor: command::Executor, n_payouts: usize) -> Self {
+        Self {
+            endpoint,
+            tasks: Tasks::default(),
+            executor,
+            n_payouts,
+        }
+    }
+}
+
+#[async_trait]
+impl xtra::Actor for Actor {
+    type Stop = ();
+
+    async fn stopped(self) -> Self::Stop {}
+}
+
+#[derive(Clone, Copy)]
+pub struct Settle {
+    pub order_id: OrderId,
+    pub price: Price,
+    pub maker_peer_id: PeerId,
+}
+
+#[xtra_productivity]
+impl Actor {
+    pub async fn handle(&mut self, msg: Settle) -> Result<()> {
+        let Settle {
+            order_id,
+            price,
+            maker_peer_id,
+        } = msg;
+
+        let (collab_settlement_tx, _) = self
```

Comment:
We could consider moving the executor into the task for the taker as well. Not super important here, but I am not sure if the feedback is actually necessary in the caller and we avoid blocking then. It's unlikely that blocking will cause issues on the taker, but still.

Reply:
I wrote multiple times in this and other tickets, that this spans 

> Please go over all the TODOs and evaluate if they are still needed.
> I think we should _not_ merge this if the maker is not able to handle previously opened positions. Can't we just try libp2p and fallback to legacy if we can't dispatch?

Thanks for double-checking: they are needed, I would not have kept them if they weren't. Please read the PR description and the comment above.

Your concern is addressed in a separate ticket added to this iteration.